### DISCUSSION TITLE
Please help me to make theme

### DISCUSSION THREAD

**Discussion by mahaveergurjar** [prasanthrangan/hyprdots#452] [Q&A] [answered]
I like mac os theme Bigsur i try to make the theme but the value are not fetch my waybar and rofi 

![231106_10h08m00s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/57872034/217a00c1-b0a9-48d4-93af-5bb6305ec6fe)

![231106_10h08m33s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/57872034/de26f679-3ca4-4ee6-83e3-7b1f70e14fca)

![231106_10h10m06s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/57872034/d6578ef7-e156-4345-946d-507e1bceed21)

**Comment by mahaveergurjar** [comment]
@T-Crypt please help me

**Reply by T-Crypt** [reply, marked as answer]
Tittu only uses like 1-2 of the default color definition names. 

Copy and paste one of the rofi theme files and fill out your colors replacing the pasted theme colors and rename it to your theme! 

Hyprpicker is a good tool to grab colors or a color hex site.

**Reply by T-Crypt** [reply]
@mahaveergurjar -- I got you. 

Give me like an hour and im making your MacOS theme in my test branch.

**Reply by mahaveergurjar** [reply]
Okay Thankyou sir

**Comment by T-Crypt** [comment]
@mahaveergurjar 

You can get your MacOS theme in my test branch on my fork of the repository or just clone it and install.sh (recommended because i added everything)

`git clone -b MacOS https://github.com/T-Crypt/hyprdots`

or 

`gh repo clone T-Crypt/hyprdots -- --branch MacOS`

then:

```
cd hyprdots/Scripts
./install.sh
```

**Reply by T-Crypt** [reply]
@mahaveergurjar -- I had some issues with the previous font package for the Icons for MacOS so if you already cloned it.  (If you haven't yet, then ignore and just install.sh as normal!)

I would delete the MacOS theme file in `~/.themes/MacOS`

and 

I would delete the MacOS icon pack in `/.icons/MacOS` 

and then run the script again!

**Reply by mahaveergurjar** [reply]
Thanks sir the are theme works perfectly but I like dark the most can you please make it dark

**Reply by T-Crypt** [reply]
Yeah I can add a feature to switch from light to dark! 

Give me like a day or so, work is pretty busy today but I'll get on this today

**Reply by mahaveergurjar** [reply]
Okay Thankyou

**Comment by T-Crypt** [comment]
[Preview]

![theme_macos_1](https://github.com/prasanthrangan/hyprdots/assets/132922589/022092b1-b11c-4ae4-b810-50e71efb2157)

![theme_macos_2](https://github.com/prasanthrangan/hyprdots/assets/132922589/3113e565-a392-47fd-b0ec-9d2672cf774e)

**Comment by kRHYME7** [comment]
Wow @T-Crypt  You're really very good at this. 
 I'm wondering if it is complicated to make a  "Patching" workflow for theme installation. Like 
```
 patch.sh /path/to/compressed/themepack
```

I know your hands are full for other projects maybe I will start to write it and I'll ask you for some advice soon?.

**Reply by prasanthrangan** [reply]
I **really** like this idea...
it will be really cool to install the themes as modules.. so that users can maintain and install their own git repo/themes to integrate with this!

like this -

```
./themepatcher.sh https://github.com/prasanthrangan/hyprdots-theme
```
where the script should,
1. clone the git theme repo
2. validates the theme files
3. copy the theme files to appropriate directories
4. update the `~/.config/swww/wall.ctl` to enable the theme

**Comment by mahaveergurjar** [comment]
@T-Crypt 
waybar not change color on the bases of wallpaper how to fix this issue

**Reply by T-Crypt** [reply]
What do you mean? Waybar isn't updating with wallbash enabled?

**Reply by mahaveergurjar** [reply]
Yes waybar is not updating with wallbash

**Reply by T-Crypt** [reply]
Can you run 

`~/.config/hypr/scripts/wallbashtoggle.sh` and share the output?

**Reply by T-Crypt** [reply]
Mine is working for me!

![231107_09h20m57s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/132922589/710904eb-3d64-40b8-ac09-03fadd1f086c)
![231107_09h21m12s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/132922589/88ef4bfd-09ca-4075-9878-6fa137785367)

**Reply by mahaveergurjar** [reply]
now working after running the script

**Reply by T-Crypt** [reply]
I also merged Windows 11 into the MacOs branch. 

Feel free to install both if you like!

**Reply by prasanthrangan** [reply]
whoaa @T-Crypt ... looks really cool!

**Reply by T-Crypt** [reply]
@prasanthrangan -- I was trying to get END-4's Windoes Eww bar working with the Windows-11 setup but its a lot of work -- the icons dont stay cached but its neat!

![231108_15h09m56s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/132922589/2c9d3022-3e4b-4016-9ef2-185300a8b00b)
![231108_15h10m25s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/132922589/3c380d15-0de1-40dd-9a9d-38b147424861)

**Comment by mahaveergurjar** [comment]
> I also merged Windows 11 into the MacOs branch.
> 
> Feel free to install both if you like!
Okay i will try 😁

**Comment by T-Crypt** [comment]
[Windows 11 Theme Preview]

![theme_windows_1](https://github.com/prasanthrangan/hyprdots/assets/132922589/4f4d69e5-9c5c-442b-85a3-dbb7835e4d03)
![theme_windows_3](https://github.com/prasanthrangan/hyprdots/assets/132922589/186d3b0a-5b89-4704-b598-1a47f565e9d8)
